cEP-0019.md

```diff
@@ -0,0 +1,379 @@
+          ]
+        },
+        "comments": {
+          "nodes": [
+            {
+              "bodyText": "To be honest, I don't understand why we need fetch-issues.js. From the relevant code, I can see fetch-issues.js does some pre-fetching and initial-data.js dumps it into the database. But what's the aim of doing that? It doesn't seem to save API hits. These pre-fetched issues in database are replaced later anyway.",
+              "author": {
+                "login": "li-boxuan",
+                "url": "https://github.com/li-boxuan"
+              },
+              "reactions": {
+                "totalCount": 0,
+                "nodes": []
+              }
+            }
+          ]
+        }
+      }
+    }
+  }
+}
+```
+
+The collection of meta-reviews includes two parts:
+
+##### Pre-fetch meta-reviews
+
+Currently, `fetch-issues.js` pre-fetches issues and pull requests to speed up
+gh-board performance and save users' API hits. Meta-reviews are also to be
+fetched during this procedure.
+
+##### Fetch meta-reviews
+
+When a new issue/pull request is detected, gh-board will fetch that issue and
+loads it asynchronously. Meta-reviews also need to be fetched during this
+procedure.
+
+### Analysis of Meta-reviews
+
+This section explains how meta-review information can be utilized to
+generate a meaningful analysis.
+
+As the collection of meta-reviews, this part also includes two parts:
+
+##### Pre-process meta-reviews
+
+After pre-fetching meta-reviews, they are to be processed, which is explained
+in the following sections.
+
+##### Process meta-reviews
+
+As new meta-reviews are loaded, useful information is to be extracted and
+current analysis of meta-review is to be updated correspondingly.
+
+The following sections explain how meta-review information is used to
+generate a meaningful analysis.
+
+#### Ranking list
+
+A ranking list would be displayed on a separate column on gh-board, according
+to points people get. A sample formula for points would be as following:
+
+PR = total number of positive reactions (e.g., THUMBS_UP) a person gets
+for all reviews he did.
+NR = total number of negative reactions (e.g., THUMBS_DOWN) a person gets
+for all reviews he did.
+PG = total number of positive reactions a person gives to other people for
+their reviews.
+NG = total number of negative reactions a person gives to other people for
+their reviews.
+
+Note PG and NG are used to encourage people to get involved in this
+reactions feedback system. Also, it may not be limited to PRs. Then final
+score S would be:
+
+S = PR - NR + c1 * PG + c2 * NG
+
+where c1 = 0.5, c2 = 0.8, are coefficients. c2 is slightly larger than c1
+because people are usually reluctant to give negative reactions.
+
+If a person has zeros for all factors, then he is regarded as an inactive
+reviewer and would not be displayed on board.
+
+#### Meta-reviews in Need
+
+Review comments that need meta-review will be displayed on a separate column
```

Comment:
When is a PR needing meta-review?  after the review comment, or after the PR is merged/closed?

The accuracy of a review comment is not known until the PR is merged or closed.

However the usefulness of a review comment to the PR author might be high , even if the review comment was not 100% correct.

So there are two, possibly three, perspectives.

meta-review by the author.

meta-review by other people in "developers" group, excluding author, (assuming developers group includes all maintainers also) where the meta-review is done before merge/close

meta-review by maintainers only, after merge/close.  These are the real gold.  They will reflect the accuracy of the review comments.

There is also the potential for reverted commits.  A reverted commit should trigger a new meta-review of the original commit, as time may have proved that the merge was incorrect (or maybe the merge was a self-aware time-sensitive workaround, and is no longer needed).

Reply:
Do you mean the gh-board should display differently according to the visitor's status (if he is a maintainer, developer, author)?